constraints/src/main/java/gov/nasa/jpl/aerie/constraints/model/LinearProfile.java

```diff
@@ -1,203 +1,134 @@
 package gov.nasa.jpl.aerie.constraints.model;
 
 import gov.nasa.jpl.aerie.constraints.time.Interval;
+import gov.nasa.jpl.aerie.constraints.time.IntervalMap;
 import gov.nasa.jpl.aerie.constraints.time.Segment;
 import gov.nasa.jpl.aerie.constraints.time.Windows;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
 import org.apache.commons.lang3.tuple.Pair;
 
 import java.util.ArrayList;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Objects;
+import java.util.Optional;
 import java.util.function.BiFunction;
 import java.util.function.Function;
 
-public final class LinearProfile implements Profile<LinearProfile> {
+public final class LinearProfile implements Profile<LinearProfile>, Iterable<Segment<LinearEquation>> {
   // IMPORTANT: Profile pieces must be non-overlapping, and increasing (based on interval field)
```

Comment:
I think this comment is now stale, since this invariant is now the responsibility of `IntervalMap`